[BRAPI]Markdown endpoint docs (#21017)
cloudflare/cloudflare-docs: src/content/docs/browser-rendering/rest-api/markdown-endpoint.mdx

---
pcx_content_type: how-to
title: Extract Markdown from a webpage
sidebar:
  order: 10
---

The `/markdown` endpoint retrieves a webpage's content and converts it into Markdown format. You can specify a URL and optional parameters to refine the extraction process.

## Basic usage

> ### Using a URL

Reviewer: ```suggestion
### Retrieve content via URL
```

This example fetches the Markdown representation of a webpage.

```bash
curl -X 'POST' 'https://api.cloudflare.com/client/v4/accounts/<accountId>/browser-rendering/markdown' \
  -H 'Content-Type: application/json' \
  -H 'Authorization: Bearer <apiToken>' \
  -d '{
    "url": "https://example.com"
  }'
```

> ### JSON response

Reviewer: ```suggestion
#### JSON response
```

```json title="json response"
{
	"success": true,
	"result": "# Example Domain\n\nThis domain is for use in illustrative examples in documents. You may use this domain in literature without prior coordination or asking for permission.\n\n[More information...](https://www.iana.org/domains/example)"
}
```

> ### Using raw HTML

Reviewer: ```suggestion
### Use raw HTML
```

> Instead of fetching a URL, you can provide raw HTML content directly.

Reviewer: ```suggestion
Instead of fetching the content by specifying the URL, you can provide raw HTML content directly.
```
Not sure if I've understood this correctly

Author: I think you have phrased it better actually.

```bash
curl -X 'POST' 'https://api.cloudflare.com/client/v4/accounts/<accountId>/browser-rendering/markdown' \
  -H 'Content-Type: application/json' \
  -H 'Authorization: Bearer <apiToken>' \
  -d '{
    "html": "<div>Hello World</div>"
  }'
```

> ### JSON response

Reviewer: ```suggestion
#### JSON response
```
